.github/styles/cloudflare/BrandNames.yml

```
@@ -0,0 +1,16 @@
+---
+# Warning: cloudflare.BrandNames
+#
+# Suggests correct spelling and capitalization for brand names.
+#
+# For a list of all options, see https://vale.sh/docs/topics/styles/
+
+extends: substitution
+message: "**Warning**: Use correct brand name: '%s' instead of '%s'."
+level: warning
+ignorecase: false
+swap:
+  (?!Cloudflare)(?:[cC]loud[fF]lare): Cloudflare
+  (?!cdnjs)(?:[cC][dD][nN]\.?[jJ][sS]): cdnjs
+  (?!GitHub)(?:[gG]it\s?[hH]ub): GitHub
+  (?!GitLab)(?:[gG]it\s?[lL]ab): GitLab
```

Thanks for putting this together Matt!

It looks like the rule currently matches strings like `create-cloudflare-cli` or `https://github.com/cloudflare`.

It's not pretty, but `(?:^|\s)` at the beginning (whitespace or start of line) and `(?:'s|\s|$)` (`'s`, whitespace or end of line) seems to prevent the false-positives.

This means we don't match `create-cloudflare-cli`, or `github.com/cloudflare`, but can still make sure we don't let `github's` or `CloudFlare's` usage slip through.

I ran the suggestion across all Markdown files in the repository and all of the warnings were valid.

Totally open to any regexes that look a little less ugly!
```suggestion
swap:
  (?:^|\s)(?!Cloudflare)(?:[cC]loud[fF]lare)(?:'s|\s|$): Cloudflare
  (?:^|\s)(?!cdnjs)(?:[cC][dD][nN]\.?[jJ][sS])(?:'s|\s|$): cdnjs
  (?:^|\s)(?!GitHub)(?:[gG]it\s?[hH]ub)(?:'s|\s|$): GitHub
  (?:^|\s)(?!GitLab)(?:[gG]it\s?[lL]ab)(?:'s|\s|$): GitLab
```

---

I think these are going to want to be lookarounds rather than non-capturing groups? Otherwise it's going to want to remove whitespace (or the `'s`) as part of the suggestion?

---

```suggestion
swap:
  (?<=^|\s)(?!Cloudflare)(?:[cC]loud[fF]lare)(?='s|\s|$): Cloudflare
  (?<=^|\s)(?!cdnjs)(?:[cC][dD][nN]\.?[jJ][sS])(?='s|\s|$): cdnjs
  (?<=^|\s)(?!GitHub)(?:[gG]it\s?[hH]ub)(?='s|\s|$): GitHub
  (?<=^|\s)(?!GitLab)(?:[gG]it\s?[lL]ab)(?='s|\s|$): GitLab
```